target_snowflake/__init__.py

```diff
@@ -170,6 +171,31 @@ def persist_lines(config, lines, table_cache=None, file_format_type: FileFormatT
             else:
                 records_to_load[stream][primary_key_string] = o['record']
 
+            if config.get('archive_load_files.enabled'):
+                # keep track of min and max
+                archive_load_files_primary_column =\
+                    config.get('archive_load_files.primary_column') or primary_key_string
```

Comment:
this should be done once when processing the `SCHEMA` message and not at every `RECORD` message.

Primary key column(s) is available in the schema message as `o['key_properties']` and the incremental key(s) as `o['bookmark_properties'].